tortoise/sim/layer.go

```diff
@@ -32,6 +33,12 @@ func WithNextReorder(delay uint32) NextOpt {
 	}
 }
 
+func WithNextBeacon(beacon []byte) NextOpt {
```

Comment:
i am not sure that this is what i planned for it, this makes a beacon a special case that depends on some external info.
i think that i definitely want to generate good beacon at Setup stage, but idk what is the best way to handle beacon changes. maybe there should be another action Beacon().

for example, i am adding action called [Partition](https://github.com/dshulyak/go-spacemesh/commit/0390a903eccb03b6817684ddcd6dc56b3d33498d#diff-9c8debd08fda2ee55a056f1dd4207fd57f10420f2a9791f7a57e7e9f36afc20c) 

the idea is that some layer on top of generator will select action that leads to state change and construct sequence of such actions randomly

Reply:
i agree. i wasn't sure how to go about this so also added a test in `TestHealing`.
i think once you mature this simulator package maybe i can tag along the `Partition` setup to have different beacon values. for now, i'll just add the test with mocked beacon